diff --git a/src/tm_case.c b/src/tm_case.c
index 2f2add878..c6a390d97 100644
--- a/src/tm_case.c
+++ b/src/tm_case.c
@@ -124,7 +124,7 @@ static void InitSelectedTMSpriteData(u8 a0, u16 itemId);
 static void SpriteCB_MoveTMSpriteInCase(struct Sprite * sprite);
 static void LoadTMTypePalettes(void);
 static void DrawPartyMonIcons(void);
-static void TintPartyMonIcons(u8 tm);
+static void TintPartyMonIcons(u16 tm);
 static void DestroyPartyMonIcons(void);
 static u8 CanTeachMove(struct Pokemon *, u16);
 
@@ -589,7 +589,7 @@ static void TMCase_MoveCursor_UpdatePrintedDescription(s32 itemIndex)
     AddTextPrinterParameterized_ColorByIndex(1, 2, str, 2, 3, 1, 0, 0, 0);
 
     // update icons
-    TintPartyMonIcons(itemId - ITEM_TM01);
+    TintPartyMonIcons(itemId);
 }
 
 static void FillBG2RowWithPalette_2timesNplus1(s32 a0)
@@ -1164,7 +1164,7 @@ static void DrawPartyMonIcons(void)
     }
 }
 
-static void TintPartyMonIcons(u8 tm)
+static void TintPartyMonIcons(u16 tm)
 {
     u8 i;
     u16 species;
@@ -1174,7 +1174,7 @@ static void TintPartyMonIcons(u8 tm)
         species = GetMonData(&gPlayerParty[i], MON_DATA_SPECIES);
         SetGpuReg(REG_OFFSET_BLDCNT, BLDCNT_TGT2_ALL);
         SetGpuReg(REG_OFFSET_BLDALPHA, BLDALPHA_BLEND(7, 11));
-        if (!CanLearnTeachableMove(species, tm)) 
+        if (!CanLearnTeachableMove(species, ItemIdToBattleMoveId(tm)))
         {
             gSprites[spriteIdData[i]].oam.objMode = ST_OAM_OBJ_BLEND;
         }
@@ -1182,10 +1182,10 @@ static void TintPartyMonIcons(u8 tm)
         {
             gSprites[spriteIdData[i]].oam.objMode = ST_OAM_OBJ_NORMAL;//gMonIconPaletteIndices[species];
         }
-
     }
 
 }
+
 static void DestroyPartyMonIcons(void)
 {
     u8 i;
